### PR TITLE
implement OGC API - Records (#404)

### PR DESCRIPTION
**Pull request by tomkralidis**
This PR implements #404 in support of the workflow of pygeoapi supporting a metadata catalogue as an OGC API collection.

This PR does not address building out a catalogue based on what is "on board", in other words, a queryable `/collections/` endpoint.  This will be put forth in another PR as things become clearer in OGC API - Records proper.

### CODE REVIEW

**Review by kalxas — APPROVED** [review]

**Review by pvgenuchten — APPROVED** [review]

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -170,7 +170,7 @@ default.
               # see pygeoapi.plugin for supported providers
               # for custom built plugins, use the import path (e.g. mypackage.provider.MyProvider)
               # see Plugins section for more information
-              - type: feature # underlying data geospatial type: (allowed values are: feature, coverage)
+              - type: feature # underlying data geospatial type: (allowed values are: feature, coverage, record)
```
Comment (francbartoli) [inline docs/source/configuration.rst:173]: Among allowed values, there should be `tile` too

File: docs/source/introduction.rst [hunk 2]
```diff
@@ -40,13 +40,15 @@ Standards are at the core of pygeoapi.  Below is the project's standards support
...
+   `OGC API - Records`_,Implementing
    `SpatioTemporal Asset Catalog`_,Implementing
 
 
 .. _`pygeoapi`: https://pygeoapi.io
 .. _`OGC API`: https://ogcapi.ogc.org
 .. _`OGC API - Features`: https://www.ogc.org/standards/ogcapi-features
-.. _`OGC API - Coverages`: https://github.com/opengeospatial/ogc_api_coverages
+.. _`OGC API - Coverages`: https://github.com/opengeospatial/ogcapi-coverages
 .. _`OGC API - Tiles`: https://github.com/opengeospatial/OGC-API-Tiles
```
Comment (francbartoli) [inline docs/source/introduction.rst:51]: it has been renamed to ogcapi-tiles

File: pygeoapi/openapi.py [hunk 3]
```diff
@@ -44,11 +44,11 @@
 
 OPENAPI_YAML = {
     'oapif': 'http://schemas.opengis.net/ogcapi/features/part1/1.0/openapi/ogcapi-features-1.yaml',  # noqa
-    'oapip': 'https://raw.githubusercontent.com/opengeospatial/wps-rest-binding/master/core/openapi',  # noqa
-#    'oacov': 'https://raw.githubusercontent.com/opengeospatial/OGC-API-Sprint-August-2020/master/docs/Draft_Spring_Guide_for_OGC_API_Coverages/openapi'  # noqa
+    'oapip': 'https://raw.githubusercontent.com/opengeospatial/ogcapi-processes/master/core/openapi',  # noqa
     'oacov': 'https://raw.githubusercontent.com/tomkralidis/ogcapi-coverages-1/fix-cis/yaml-unresolved',  # noqa
     'oapit': 'https://raw.githubusercontent.com/opengeospatial/OGC-API-Tiles/master/openapi/swaggerhub/tiles.yaml',  # noqa
```
Comment (francbartoli) [inline pygeoapi/openapi.py:49]: OGC-API-Tiles --> ogcapi-tiles

**Review by francbartoli — APPROVED** [review]

**Comment by tomkralidis** [comment]
Test failures unrelated.

**Pull request merged**